### PR TITLE
fix: Add new linter copyloopvar to prevent copying local variables in loop

### PR DESCRIPTION
**Pull request by Matovidlo**
Jira: [PSGO-638](https://keboola.atlassian.net/browse/PSGO-638)

**Changes:**
- Simplifies PR reviews.

-----------

[PSGO-638]: https://keboola.atlassian.net/browse/PSGO-638?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: internal/pkg/service/stream/storage/repository/file_repo.go [hunk 1]
```diff
@@ -177,7 +177,7 @@ func (r *FileRepository) StateTransition(now time.Time, fileKey model.FileKey, f
 			txn := op.Txn(r.client)
 			errs := errors.NewMultiError()
 			for _, slice := range fileSlices {
-				oldSliceState := slice
+				oldSliceState := slice // nolint: copyloopvar
```
Comment (Matovidlo) [inline internal/pkg/service/stream/storage/repository/file_repo.go:180]: Could this be refactored?

File: internal/pkg/template/repository/manager/manager.go [hunk 2]
```diff
@@ -169,7 +167,7 @@ func (m *Manager) Update(ctx context.Context) <-chan error {
 				m.repositoriesLock.Unlock()
 
 				// Free previous value
-				oldValue.free(ctx)
+				repo.free(ctx)
```
Comment (Matovidlo) [inline internal/pkg/template/repository/manager/manager.go:170]: Not sure about freeing loop var.

File: internal/pkg/service/stream/storage/repository/file_repo.go (hunk 1 above)
Comment (michaljurecko) [reply internal/pkg/service/stream/storage/repository/file_repo.go:180]: yes, it is in the API PRs

File: internal/pkg/service/appsproxy/dataapps/appconfig/appconfig_test.go [hunk 3]
```diff
@@ -223,7 +223,6 @@ func TestLoader_LoadConfig(t *testing.T) {
 	}
 
 	for _, tc := range testCases {
-		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
```
Comment (michaljurecko) [inline internal/pkg/service/appsproxy/dataapps/appconfig/appconfig_test.go:226]: I already removed it in API PRs, so please rebase and merge it, after the API part will be merged, at first.

**Review by michaljurecko — APPROVED** [review]

**Pull request closed without merging**